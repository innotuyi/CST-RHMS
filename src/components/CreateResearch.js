import React, { useState } from "react";
import axios from 'axios';
import { useHistory } from "react-router-dom";
import {
  Row,
  Col,
  Container,
  Form,
  Button,
  Jumbotron,
} from "react-bootstrap";
import Header from "./Header";

function CreateResearch() {

  const history = useHistory();

 
  const [title, setTitle] = useState("");
  const [startDate, setStartDate] = useState("");
  const [endingDate, setEndingDate] = useState("");
  const [amount, setAmount] = useState("");
  const [status, setStatus] = useState("");
  const [type, setType] = useState("");
  // const [dateCreated, setDateCreated] = useState("");

  const submitHandler = (e) => {
    e.preventDefault();
    if(type=="Chapter") {
      const payload = {
        'research_title': title, 
        'starting_date': setStartDate,
        'ending_date':endingDate, 
         'amount':amount,
         'status':status,
         // 'date_created':dateCreated,
          'owner':1}

axios.post('http://127.0.0.1:8000/research/', payload)
.then(function (response) {
console.log('Successful response: ',  response);
})
.catch(function (error) {
console.log('Error response: ', error);
});

    
history.push('/chapterbased');
    }
    else if(type=="Conference") {
      const payload = {
        'research_title': title, 
          'starting_date': setStartDate,
          'ending_date':endingDate, 
         'amount':amount,
         'status':status,
         // 'date_created':dateCreated,
          'owner':1}

axios.post('http://127.0.0.1:8000/research/', payload)
.then(function (response) {
console.log('Successful response: ',  response);
})
.catch(function (error) {
console.log('Error response: ', error);
});
      history.push('/ConferenceBased');

    }
    else if(type=="Book") {
      const payload = {
        'research_title': title, 
          'starting_date': setStartDate,
          'ending_date':endingDate, 
         'amount':amount,
         'status':status,
         // 'date_created':dateCreated,
          'owner':1}

axios.post('http://127.0.0.1:8000/research/', payload)
.then(function (response) {
console.log('Successful response: ',  response);
})
.catch(function (error) {
console.log('Error response: ', error);
});
      history.push('/bookbased');
     
      
    }

    else if(type=="Jounal") {
      const payload = {
        'research_title': title, 
          'starting_date': setStartDate,
          'ending_date':endingDate, 
         'amount':amount,
         'status':status,
         // 'date_created':dateCreated,
          'owner':1}

axios.post('http://127.0.0.1:8000/research/', payload)
.then(function (response) {
console.log('Successful response: ',  response);
})
.catch(function (error) {
console.log('Error response: ', error);
});
      history.push('/JournalBased');
      
    }
    else  {

      console.warn(title,amount, status);

    }
    
   };


  return (
    <div
      style={{
        background: "linear-gradient(to bottom, white, 84%, #1089AF)",
        height:"100vh"
      }}
    >
      <Header />
      <Container>
        <Row className="d-flex justify-content-center arighn-item-center">
          <Col className="d-flex justify-content-center">
            <Jumbotron
              style={{
                background: "linear-gradient(to bottom, white , 90%, #1089AF )",
                borderRadius: "10px",
                width: "90%",
              }}
            >
              <Form className="form" style={{}} onSubmit={submitHandler}>
                <h2 className="text-center mb-5 text-secondary">
                  Create New Research
                </h2>
                <Form.Group>
                  <Form.Label>Title</Form.Label>
                  <Form.Control
                    required
                    type="text"
                    id="title"
                    value={title}
                    onChange={(e) => setTitle(e.target.value)}
                    placeholder="Enter Research Title"
                  />
                </Form.Group>
                <Row>
                  <Col>
                    <Form.Group controlId="dob">
                      <Form.Label>Select Starting Date</Form.Label>
                      <Form.Control
                        type="date"
                        name="dos"
                        required
                        id="startdate"
                        value={startDate}
                        onChange={(e) => setStartDate(e.target.value)}
                        placeholder="Starting Date"
                      />
                    </Form.Group>
                  </Col>
                  <Col>
                    <Form.Group>
                      <Form.Label>Select Ending Date</Form.Label>
                      <Form.Control
                        type="date"
                        name="dos"
                        required
                        id="endingdate"
                        value={endingDate}
                        onChange={(e) => setEndingDate(e.target.value)}
                        placeholder="Ending Date"
                      />
                    </Form.Group>
                  </Col>
                </Row>
                 <Form.Group>
                  <Form.Label>Project Description</Form.Label>
                  <Form.Control
                    required
                    type="text"
                    id="amount"
                    value={amount}
                    onChange={(e) => setAmount(e.target.value)}
                    placeholder="Enter  Project Description"
                  />
                </Form.Group>
                <Form.Group controlId="">
                  <Form.Label>Select Research Status</Form.Label>
                  <Form.Control
                    as="select"
                    required
                    id="status"
                    value={status}
                    onChange={(e) => setStatus(e.target.value)}
                    
                  >
                   
                    <option value="ongoing">Ongoing</option>
                    <option value="Completed">Completed</option>
                   
                  </Form.Control>
                </Form.Group>
                {/* <Form.Group>
                  <Form.Label>Status</Form.Label>
                  <Form.Control
                    required
                    type="text"
                    id="status"
                    value={status}
                    onChange={(e) => setStatus(e.target.value)}
                    placeholder="Enter Research status"
                  />
                </Form.Group> */}

                
{/* 
                <Row>
                  <Col>
                    <Form.Group controlId="dob">
                      <Form.Label>Select Date Created</Form.Label>
                      <Form.Control
                        type="date"
                        name="dos"
                        required
                        id="dateCreated"
                        value={dateCreated}
                        onChange={(e) => setDateCreated(e.target.value)}
                        placeholder="Date Createdd"
                      />
                    </Form.Group>
                  </Col>
                </Row> */}
                <Form.Group controlId="">
                  <Form.Label>Select Research Expected Outcome</Form.Label>
                  <Form.Control
                    as="select"
                    required
                    id="type"
                    value={type}
                    onChange={(e) => setType(e.target.value)}
                   
                  >
                    <option value="Jounal">Jounal Based Research</option>
                    <option value="Conference">Conference Based Research</option>
                    <option value="Book">Book Based Research</option>
                    <option value="Chapter">Chapter Based Research</option>
                    <option></option>
                  </Form.Control>
                </Form.Group>
                <Col className="text-center">
                  <Button className="" type="submit">
                    Create Research
                  </Button>
                </Col>
              </Form>
            </Jumbotron>
          </Col>
        </Row>
      </Container>
    </div>
  );
}

export default CreateResearch;
