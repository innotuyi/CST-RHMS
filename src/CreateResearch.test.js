import React from "react";
import CreateResearch from "./components/CreateResearch";

import { configure, shallow, mount } from "enzyme";
import toJSON from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

let wrapper;
beforeEach(() => {
  wrapper = shallow(<CreateResearch />);
});

describe("Create Research Tests", () => {
  it("matches Snapshot Collectly", () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

   it("should have All Inputs", () => {
     expect(wrapper.find("#title")).toHaveLength(1);
     expect(wrapper.find("#startdate")).toHaveLength(1);
     expect(wrapper.find("#endingdate")).toHaveLength(1);
     expect(wrapper.find("#amount")).toHaveLength(1);
     expect(wrapper.find("#type")).toHaveLength(1);
   });

   it("should have a Button  Component", () => {
     expect(wrapper.find("Button")).toHaveLength(1);
     expect(wrapper.find("Button").text()).toEqual("Create Research");
   });

   it("Should capture Research Title correctly onChange", () => {
     wrapper.find("#title").simulate("change", {
       target: {
         value: "New Title",
       },
     });
     expect(wrapper.find("#title").props().value).toEqual("New Title");
   });

   it("Should capture Starting Date correctly onChange", () => {
     wrapper.find("#startdate").simulate("change", {
       target: {
         value: "2021-06-09",
       },
     });
     expect(wrapper.find("#startdate").props().value).toEqual("2021-06-09");
   });

   it("Should capture Ending Date correctly onChange", () => {
     wrapper.find("#endingdate").simulate("change", {
       target: {
         value: "2021-05-19",
       },
     });
     expect(wrapper.find("#endingdate").props().value).toEqual("2021-05-19");
   });
  
  it("Should capture Amount correctly onChange", () => {
    wrapper.find("#amount").simulate("change", {
      target: {
        value: "800",
      },
    });
    expect(wrapper.find("#amount").props().value).toEqual("800");
  });

  it("Should capture Research Type correctly onChange", () => {
    wrapper.find("#type").simulate("change", {
      target: {
        value: "Jounal Based Research",
      },
    });
    expect(wrapper.find("#type").props().value).toEqual("Jounal Based Research");
  });

});
