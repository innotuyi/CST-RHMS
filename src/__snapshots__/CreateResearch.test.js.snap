// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Research Tests matches Snapshot Collectly 1`] = `
<div
  style={
    Object {
      "background": "linear-gradient(to bottom, white, 84%, #1089AF)",
      "height": "100vh",
    }
  }
>
  <Header />
  <Container
    fluid={false}
  >
    <Row
      className="d-flex justify-content-center arighn-item-center"
      noGutters={false}
    >
      <Col
        className="d-flex justify-content-center"
      >
        <Jumbotron
          fluid={false}
          style={
            Object {
              "background": "linear-gradient(to bottom, white , 90%, #1089AF )",
              "borderRadius": "10px",
              "width": "90%",
            }
          }
        >
          <Form
            className="form"
            inline={false}
            onSubmit={[Function]}
            style={Object {}}
          >
            <h2
              className="text-center mb-5 text-secondary"
            >
              Create New Research
            </h2>
            <FormGroup>
              <FormLabel
                column={false}
                srOnly={false}
              >
                Title
              </FormLabel>
              <FormControl
                id="title"
                onChange={[Function]}
                placeholder="Enter Research Title"
                required={true}
                type="text"
                value=""
              />
            </FormGroup>
            <Row
              noGutters={false}
            >
              <Col>
                <FormGroup
                  controlId="dob"
                >
                  <FormLabel
                    column={false}
                    srOnly={false}
                  >
                    Select Starting Date
                  </FormLabel>
                  <FormControl
                    id="startdate"
                    name="dos"
                    onChange={[Function]}
                    placeholder="Starting Date"
                    required={true}
                    type="date"
                    value=""
                  />
                </FormGroup>
              </Col>
              <Col>
                <FormGroup>
                  <FormLabel
                    column={false}
                    srOnly={false}
                  >
                    Select Ending Date
                  </FormLabel>
                  <FormControl
                    id="endingdate"
                    name="dos"
                    onChange={[Function]}
                    placeholder="Ending Date"
                    required={true}
                    type="date"
                    value=""
                  />
                </FormGroup>
              </Col>
            </Row>
            <FormGroup>
              <FormLabel
                column={false}
                srOnly={false}
              >
                Amount
              </FormLabel>
              <FormControl
                id="amount"
                onChange={[Function]}
                placeholder="Enter Research Amount"
                required={true}
                type="number"
                value=""
              />
            </FormGroup>
            <FormGroup
              controlId=""
            >
              <FormLabel
                column={false}
                srOnly={false}
              >
                Select Research Type
              </FormLabel>
              <FormControl
                as="select"
                defaultValue="Select Research Type ..."
                id="type"
                onChange={[Function]}
                required={true}
                value=""
              >
                <option>
                  Select Research Type...
                </option>
                <option>
                  Jounal Based Research
                </option>
                <option>
                  Conference Based Research
                </option>
                <option>
                  Book Based Research
                </option>
                <option>
                  Chapter Based Research
                </option>
                <option />
              </FormControl>
            </FormGroup>
            <Col
              className="text-center"
            >
              <Button
                active={false}
                className=""
                disabled={false}
                type="submit"
                variant="primary"
              >
                Create Research
              </Button>
            </Col>
          </Form>
        </Jumbotron>
      </Col>
    </Row>
  </Container>
</div>
`;
